### PR TITLE
RSDK-6058 Deprecate sensors service api

### PR DESCRIPTION
**Pull request by randhid**
This pull request marks the sensors service API as deprecated. This will be removed once the RC card for sensors no longer has a dependency on the GetAllReadings Function.

### CODE REVIEW

File: proto/viam/service/sensors/v1/sensors.proto [hunk 1]
```diff
@@ -10,49 +10,54 @@ option go_package = "go.viam.com/api/service/sensors/v1";
...
       post: "/viam/api/v1/service/sensors/{name}/do_command"
     };
+    option deprecated = true;
   }
 }
 
+// The sensor service messages are deprecated
 message GetSensorsRequest {
-  string name = 1;
+  string name = 1 [deprecated = true];
```
Comment (penguinland) [inline proto/viam/service/sensors/v1/sensors.proto:42]: This `name` is marked deprecated, but the names on lines 52 and 59 are not deprecated. Is that on purpose?

File: proto/viam/service/sensors/v1/sensors.proto [hunk 2]
```diff
@@ -10,49 +10,54 @@ option go_package = "go.viam.com/api/service/sensors/v1";
...
   // Additional arguments to the method
-  google.protobuf.Struct extra = 99;
+  google.protobuf.Struct extra = 99 [deprecated = true];
 }
 
 message GetSensorsResponse {
-  repeated common.v1.ResourceName sensor_names = 1;
```
Comment (penguinland) [inline proto/viam/service/sensors/v1/sensors.proto:43]: I think that in addition to giving RPCs the deprecated option, [you can do that to entire messages](https://github.com/protocolbuffers/protobuf/issues/1734#issuecomment-234681729), too. Worth doing here? I dunno; what you've got is already pretty thorough.

File: proto/viam/service/sensors/v1/sensors.proto [hunk 3]
```diff
@@ -10,49 +10,54 @@ option go_package = "go.viam.com/api/service/sensors/v1";
...
+  google.protobuf.Struct extra = 99 [deprecated = true];
 }
 
 message Readings {
   common.v1.ResourceName name = 1;
-  map<string, google.protobuf.Value> readings = 2;
+  map<string, google.protobuf.Value> readings = 2 [deprecated = true];
 }
 
 message GetReadingsResponse {
```
Comment (penguinland) [inline proto/viam/service/sensors/v1/sensors.proto:63]: Was this one supposed to be deprecated, too? I had expected `GetSensorsResponse` and `GetReadingsResponse` to have the same fate.

**Review by penguinland — APPROVED** [review]
LGTM. It's exciting to get rid of this!

File: proto/viam/service/sensors/v1/sensors.proto (hunk 3 above)
Comment (randhid) [reply proto/viam/service/sensors/v1/sensors.proto:63]: ah you're too fast, was self-reviewing because I was sure I missed some.

Comment (randhid) [reply proto/viam/service/sensors/v1/sensors.proto:63]: Added now.

File: proto/viam/service/sensors/v1/sensors.proto (hunk 2 above)
Comment (randhid) [reply proto/viam/service/sensors/v1/sensors.proto:43]: nice! I am not a proto expert, had to google what to actually do here. I don't see the harm in just adding deprecated signifiers everywhere we can.

Comment (penguinland) [reply proto/viam/service/sensors/v1/sensors.proto:43]: I think it's:
```
message GetSensorsResponse {
    option deprecated = true;
    repeated common.v1.ResourceName sensor_names = 1 [deprecated = true]; 
    ...
}
```
but I've never done this myself and could be wrong.

**Review by penguinland — APPROVED** [review]

File: proto/viam/service/sensors/v1/sensors.proto (hunk 2 above)
Comment (randhid) [reply proto/viam/service/sensors/v1/sensors.proto:43]: Okay I'm satisfied. I will check with a local version of this API with rdk tomorrow to make sure things work as expected and won't break anything juts by adding a deprecated signifier to proto.

**Comment by randhid** [comment]
> This pull request marks the sensors service API as deprecated. This will be removed once the RC card for sensors no longer has a dependency on the GetAllReadings Function.

Finally taste with a local build of the API following instructions [here](https://viam.atlassian.net/wiki/spaces/ENG/pages/160989185/How+to+Run+Everything+Locally+All+At+Once#Api), in case anyone was wondering about the method I used to test.

Tested with the python SDK using the following code:
```python
    service = SensorsClient.from_robot(robot, "builtin")
    sensors = await service.get_sensors()
    print(f"b get_readings return value: {sensors}")

    vals = await service.get_readings(sensors)
    print(f"b get_readings return value: {vals}")
```

Got back:
```
get_sensors return value: [<viam.proto.common.ResourceName rdk:component:sensor/b at 0x103ee7b30>, <viam.proto.common.ResourceName rdk:component:sensor/c at 0x103ee76d0>, <viam.proto.common.ResourceName rdk:component:sensor/a at 0x103ee7680>]
get_readings return value: {<viam.proto.common.ResourceName rdk:component:sensor/b at 0x103ee7a40>: {'a': 1.0, 'b': 2.0, 'c': 3.0}, <viam.proto.common.ResourceName rdk:component:sensor/c at 0x103ee7c20>: {'a': 1.0, 'b': 2.0, 'c': 3.0}, <viam.proto.common.ResourceName rdk:component:sensor/a at 0x103ee71d0>: {'a': 1.0, 'b': 2.0, 'c': 3.0}}
```

RC Card is still functional (three fake sensors configured):

Screenshot 2024-01-12 at 17 47 43

I'm happy, will merge in.

**Review by penguinland — APPROVED** [review]
Rand thinks the PR is stuck in some weird CI edge case, and needs another LGTM to get unstuck...

**Pull request merged**